index.html

```diff
@@ -1028,6 +1061,461 @@ <h2>
+                  <ol>
+                    <li>[=Resolve=]
+                    [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+                    with {{GamepadHapticsResult/"complete"}}.
+                    </li>
+                    <li>Set
+                    [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+                    to `undefined`.
+                    </li>
+                  </ol>
+                </li>
+              </ol>
+            </li>
+            <li>Return {{GamepadHapticActuator/[[playingEffectPromise]]}}.
+            </li>
+          </ol>
+        </dd>
+        <dt>
+          <dfn>reset()</dfn>
+        </dt>
+        <dd>
+          <p>
+            The {{GamepadHapticActuator/reset()}} method steps are:
+          </p>
+          <ol>
+            <li>Let |document| be the [=current settings object=]'s [=relevant
+            global object=]'s [=associated `Document`=].
+            </li>
+            <li>If |document| is `null` or |document| is not [=Document/fully
+            active=] or |document|'s [=Document/visibility state=] is `"hidden"`, return
+            [=a promise resolved with=] {{GamepadHapticsResult/"preempted"}}.
+            </li>
+            <li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+            is not `undefined`:
+              <ol>
+                <li>Let |effectPromise| be
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}.
+                </li>
+                <li>Set
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
+                `undefined`.
+                </li>
+                <li>[=Queue a global task=] on the [=relevant global object=]
+                of [=this=] using the [=gamepad task source=] to [=resolve=]
+                |effectPromise| with {{GamepadHapticsResult/"preempted"}}.
+                </li>
+              </ol>
+            </li>
+            <li>[=Stop haptic effects=] on this actuator.
+            </li>
+            <li>Return [=a promise resolved with=]
+            {{GamepadHapticsResult/"complete"}};
+            </li>
+          </ol>
+        </dd>
+      </dl>
+      <p>
+        A {{GamepadHapticActuator}} can <dfn>play effects with type</dfn>
+        |type:GamepadHapticEffectType| if |type:GamepadHapticEffectType| can be
+        found in the {{GamepadHapticActuator/[[effects]]}} [=list=].
+      </p>
+      <p>
+        An effect with {{GamepadHapticEffectType}}
+        |type:GamepadHapticEffectType| and {{GamepadEffectParameters}}
+        |params:GamepadEffectParameters| describes a <dfn>valid effect</dfn> if
+        |params| contains all of the required parameters for |type| effects and
+        all of the parameters have valid values. Additional parameters MUST be
+        ignored.
+      </p>
+      <p>
+        A [=user agent=] that implements this API MUST provide a method to
```

Comment:
This isn't right... The should just define should define what  `<dfn>issue a haptic effect</dfn>` with the parameters means, not that it's a MUST. The MUST part is already part of playEffect(), this just needs to say what to do "i.e., make the actuator render the |type| and try to make it use the |params|" (needs better wording, but you get the idea).

Reply:
Great! I have removed this "MUST" part from here and from the "stop haptics effects" part. Also reworded it to be more cohesive.